### PR TITLE
Move QgsMapToolCapture::CaptureTechnique enum to Qgis

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

File: src/core/qgis.h [hunk 1]
```diff
@@ -136,6 +136,20 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( DataType )
 
+    /**
+     * Capture technique.
+     *
+     * \since QGIS 3.26
+     */
+    enum class CaptureTechnique SIP_MONKEYPATCH_SCOPEENUM_UNNEST( Qgis, PythonMacroMode ) : int
```
Comment (nyalldawson) [inline src/core/qgis.h:144]:
```suggestion
    enum class CaptureTechnique SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsMapToolCapture, CaptureTechnique ) : int
```

File: src/core/qgis.h [hunk 2]
```diff
@@ -136,6 +144,21 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( DataType )
 
+    /**
+     * Capture technique.
+     *
+     * \since QGIS 3.26
+     */
+    enum class CaptureTechnique SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsMapToolCapture, Qgis::CaptureTechnique ) : int
```
[the remark this answers is not in the document]
Comment (3nids) [reply src/core/qgis.h]: thanks

**Comment by jef-n** [comment]
The windows nightly is broken for a while too - https://cdash.orfeo-toolbox.org/viewBuildError.php?buildid=93967 :(

**Comment by 3nids** [comment]
@jef-n are the windows build using a different version of sip?

could you try this:
```
--- a/python/core/auto_generated/settings/qgssettingsentry.sip.in
+++ b/python/core/auto_generated/settings/qgssettingsentry.sip.in
@@ -284,9 +284,6 @@ The ``description`` argument specifies a description for the settings entry.
 The ``options`` arguments specifies the options for the settings entry.
 %End
 
-
-    virtual Qgis::SettingsType settingsType() const = 0;
-
     T value( const QString &dynamicKeyPart = QString() ) const;
 %Docstring
 Returns settings value.
@@ -416,8 +413,6 @@ The ``description`` argument specifies a description for the settings entry.
 The ``options`` arguments specifies the options for the settings entry.
 %End
 
-    virtual Qgis::SettingsType settingsType() const = 0;
-
     T value( const QString &dynamicKeyPart = QString() ) const;
 %Docstring
 Returns settings value.

```

**Comment by 3nids** [comment]
MingW builds have been fixed in other PR https://github.com/qgis/QGIS/pull/47719
@jef-n I have good hope that the windows builds have been fixed at the same time

I'd still merge the move of the enum to QGIS.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**